pkg/libvirt/resolve_vm.go

```diff
@@ -99,3 +111,49 @@ func addToNotExistCache(pid uint64) {
 	}
 	cacheNotExist[pid] = false
 }
+
+func getMetadataVMID(vmid string) (string, error) {
+	metadataURI := config.GetLibvirtMetadataURI()
+	token := config.GetLibvirtMetadataToken()
+	domainName := ""
+
+	parts := strings.Split(vmid, "-")
+	if len(parts) < 3 {
+		return "", fmt.Errorf("the vmid provided doest not match machine-qemu-number-name : %v", vmid)
+	}
+
+	domainName = strings.Join(parts[3:], "-")
```

Comment:
wouldn't this crash if len(parts) == 3 ?

Reply:
```
vmids := []string{
		"machine-qemu-1234-name",
		"machine---5678-anothername",
		"machine-qemu-91011-differentname",
		"invalid-vmid",
		"machine-0000-lastname",
		"qemu-machine-",
		"qemu-machine--123456-last-",
	}

	for _, vmid := range vmids {
		parts := strings.Split(vmid, "-")
		if len(parts) < 4 {
			fmt.Printf("The vmid provided does not match machine-qemu-number-name: %v\n", vmid)
			continue
		}

		domainName := strings.Join(parts[3:], "-")
		fmt.Println(domainName)
	}
```
Output:
```
name
5678-anothername
differentname
The vmid provided does not match machine-qemu-number-name: invalid-vmid
The vmid provided does not match machine-qemu-number-name: machine-0000-lastname
The vmid provided does not match machine-qemu-number-name: qemu-machine-
123456-last-
```

Any suggestion of name that could be problematic ?

Reply:
@aharivel the snipped you pasted, has len(parts) < 4 instead of 3 (as in the code).

Reply:
I literally have no idea what happen there but thanks for pointing that out.